src/menu/main_menu.rs

```diff
@@ -0,0 +1,162 @@
+use bevy::prelude::*;
+
+use crate::menu::{ClickAction, MenuMaterials, StateCleanup};
+use crate::AppState;
+
+pub fn setup(
+    commands: &mut Commands,
+    asset_server: Res<AssetServer>,
+    materials: Res<MenuMaterials>,
+) {
+    let button_style = Style {
+        size: Size::new(Val::Auto, Val::Auto),
+        margin: Rect::all(Val::Px(5.0)),
+        padding: Rect {
+            left: Val::Px(12.0),
+            right: Val::Px(12.0),
+            top: Val::Px(10.0),
+            bottom: Val::Px(10.0),
+        },
+        justify_content: JustifyContent::Center,
+        align_items: AlignItems::Center,
+        ..Default::default()
+    };
+
+    let button_text_style = TextStyle {
+        font: asset_server.load("fonts/sansation/sansation_light.ttf"),
+        font_size: 18.0,
+        color: Color::WHITE,
+    };
+
+    commands
+        .spawn(UiCameraBundle::default())
```

Comment:
Hmm, we should probably make the UI camera state independent - we'll have _some_ UI in all states, and this avoids a nasty workaround

Reply:
yes agreed, i think we should just spawn the UI camera at app launch (you can even do it in a start up system outside of states), and just keep it alive forever.

There is no harm, we are probably gonna have lots of UIs in every game state anyway.

It will just serve to make sure that it can always be displayed.

Reply:
Sounds good. I've replaced the UI cameras with a single global UI camera.